cmd/cca/verify_as_attester.go

```diff
@@ -0,0 +1,174 @@
+interaction, where the verifier is the protocol challenger.  Therefore, the nonce or 
+challenge is provided by the Veraison API server and the CCA attestation token needs
+to be created on the fly based on the attester's claims and platform signing key (PAK)
+and realm signing key (RAK).
+	
+	evcli cca verify-as attester \
+	              --api-server=https://veraison.example/challenge-response/v1/newSession \
+	              --claims=claims.json \
+	              --pak=pak.jwk \
+			      --rak=rak.jwk
+				   
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			ev, err := loadCCAClaimsFromFile(fs, *attesterClaimsFile)
+			if err != nil {
+				return err
+			}
+
+			key, err := afero.ReadFile(fs, *platformKeyFile)
+			if err != nil {
+				return fmt.Errorf("error loading signing key from %s: %w", *platformKeyFile, err)
+			}
+
+			platSigner, err := common.SignerFromJWK(key)
+			if err != nil {
+				return fmt.Errorf("error decoding signing key from %s: %w", *platformKeyFile, err)
+			}
+
+			key, err = afero.ReadFile(fs, *realmKeyFile)
+			if err != nil {
+				return fmt.Errorf("error loading signing key from %s: %w", *realmKeyFile, err)
+			}
+
+			realmSigner, err := common.SignerFromJWK(key)
+			if err != nil {
+				return fmt.Errorf("error decoding signing key from %s: %w", *platformKeyFile, err)
+			}
+
+			eb := attesterEvidenceBuilder{Pclaims: ev.PlatformClaims, Rclaims: ev.RealmClaims, Psigner: platSigner, Rsigner: realmSigner}
+
+			if err = attesterVeraisonClient.SetEvidenceBuilder(eb); err != nil {
+				return err
+			}
+
+			if err = attesterVeraisonClient.SetSessionURI(*attesterAPIURL); err != nil {
+				return err
+			}
+
+			if err = attesterVeraisonClient.SetNonceSz(attesterNonceSz); err != nil {
+				return err
+			}
+
+			attesterVeraisonClient.SetDeleteSession(true)
+			attestationResults, err := attesterVeraisonClient.Run()
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(string(attestationResults))
+
+			return nil
+		},
+	}
+
+	attesterClaimsFile = cmd.Flags().StringP(
+		"claims", "c", "", "JSON file containing the CCA attestation claims to be signed",
+	)
+
+	platformKeyFile = cmd.Flags().StringP(
+		"pak", "p", "", "JWK file with the Platform Attestation Key used for signing",
+	)
+
+	realmKeyFile = cmd.Flags().StringP(
+		"rak", "r", "", "JWK file with the Realm Attestation Key used for signing",
+	)
+
+	attesterAPIURL = cmd.Flags().StringP(
+		"api-server", "s", "", "URL of the Veraison verification API",
+	)
+
+	return cmd
+}
+
+func (eb attesterEvidenceBuilder) BuildEvidence(nonce []byte, accept []string) ([]byte, string, error) {
+	for _, ct := range accept {
+		if ct != CCATokenMediaType {
+			continue
+		}
+
+		if err := eb.Rclaims.SetChallenge(nonce); err != nil {
+			return nil, "", fmt.Errorf("setting nonce: %w", err)
+		}
+
+		_, err := eb.Pclaims.GetProfile()
+		if err != nil {
+			return nil, "", fmt.Errorf("getting profile: %w", err)
+		}
```

Comment:
just a curiosity: why is this needed?

Reply:
This is just a sanity check that the Evidence Builder Claim has a valid profile else we should bail out in absence of a profile!

Reply:
(just to be sure I understand) wouldn't that be caught anyway later?